In [1]:
%run -i '../../lib/fetch-data.py'

Data already downloaded, skipping...


# Privacy Analysis

In [2]:
import pandas as pd

In [28]:
df = pd.read_csv('../../data/AllUsers_Ads_Ratings_df.csv')

/tmp/ipykernel_32147/2270874857.py:1: DtypeWarning: Columns (22,28,29,30,36) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv('../../data/AllUsers_Ads_Ratings_df.csv')


In [4]:
df.head()

UserId    AdId  Age Cap/Zip-Code         Countries visited  \
0  U0001  A01_01   62        15613  United States of America   
1  U0001  A01_02   62        15613  United States of America   
2  U0001  A01_03   62        15613  United States of America   
3  U0001  A01_04   62        15613  United States of America   
4  U0001  A01_05   62        15613  United States of America   

            Fave Sports Gender              Home country Home town  Income  \
0  I do not like Sports      F  United States of America    Apollo       1   
1  I do not like Sports      F  United States of America    Apollo       1   
2  I do not like Sports      F  United States of America    Apollo       1   
3  I do not like Sports      F  United States of America    Apollo       1   
4  I do not like Sports      F  United States of America    Apollo       1   

   ... fave8 fave9         unfave1       unfave2   unfave3 unfave4  \
0  ...   NaN   NaN  news headlines  homelessness  violence     war   
1  ...   NaN   NaN  news headlines  homelessness  violence     war   
2  ...   NaN   NaN  news headlines  homelessness  violence     war   
3  ...   NaN   NaN  news headlines  homelessness  violence     war   
4  ...   NaN   NaN  news headlines  homelessness  violence     war   

        unfave5 unfave6                                         AdFilePath  \
0  human rights     NaN  ../../data/ads16-dataset/ADS16_Benchmark_part1...   
1  human rights     NaN  ../../data/ads16-dataset/ADS16_Benchmark_part1...   
2  human rights     NaN  ../../data/ads16-dataset/ADS16_Benchmark_part1...   
3  human rights     NaN  ../../data/ads16-dataset/ADS16_Benchmark_part1...   
4  human rights     NaN  ../../data/ads16-dataset/ADS16_Benchmark_part1...   

  Rating  
0    1.0  
1    1.0  
2    1.0  
3    1.0  
4    1.0  

[5 rows x 39 columns]

In [5]:
df.columns

Index(['UserId', 'AdId', 'Age', 'Cap/Zip-Code', 'Countries visited',
       'Fave Sports', 'Gender', 'Home country', 'Home town', 'Income',
       'Last Name', 'Most listened musics', 'Most read books',
       'Most visited websites', 'Most watched movies',
       'Most watched tv programmes', 'Name', 'Paypal', 'Timepass',
       'Type of Job', 'Weekly working hours', 'fave1', 'fave10', 'fave2',
       'fave3', 'fave4', 'fave5', 'fave6', 'fave7', 'fave8', 'fave9',
       'unfave1', 'unfave2', 'unfave3', 'unfave4', 'unfave5', 'unfave6',
       'AdFilePath', 'Rating'],
      dtype='object')

In [6]:
df['Cap/Zip-Code']

0        15613
1        15613
2        15613
3        15613
4        15613
         ...  
36115    60611
36116    60611
36117    60611
36118    60611
36119    60611
Name: Cap/Zip-Code, Length: 36120, dtype: object

In [7]:
df.groupby(['Cap/Zip-Code']).size()

Cap/Zip-Code
10000       301
10003       301
10009       301
10022       301
10456       301
           ... 
SS2 5JL     301
SW14 8DP    301
T3L2Z2      301
TW9 4JF     301
t6c 3e8     301
Length: 115, dtype: int64

Como temos 115 zip codes pra 120 usuários, é melhor droparmos a coluna de zip-code (ela não deve agregar em nada na predição). O mesmo vale para país e cidade e pras colunas que já estavam removidas (nome, lastname, paypal)

In [29]:
df = df.drop(['Cap/Zip-Code', 'Home country', 'Home town', 'Name', 'Last Name', 'Paypal'], axis=1)

In [9]:
df.columns

Index(['UserId', 'AdId', 'Age', 'Countries visited', 'Fave Sports', 'Gender',
       'Income', 'Most listened musics', 'Most read books',
       'Most visited websites', 'Most watched movies',
       'Most watched tv programmes', 'Timepass', 'Type of Job',
       'Weekly working hours', 'fave1', 'fave10', 'fave2', 'fave3', 'fave4',
       'fave5', 'fave6', 'fave7', 'fave8', 'fave9', 'unfave1', 'unfave2',
       'unfave3', 'unfave4', 'unfave5', 'unfave6', 'AdFilePath', 'Rating'],
      dtype='object')

In [10]:
df.to_csv("../../data/AllUsers_Ads_Ratings_Privacy_Safe.csv", index=False)

In [11]:
df_age = df.groupby(['UserId', 'Age']).count()

In [12]:
df_age.head()

AdId  Countries visited  Fave Sports  Gender  Income  \
UserId Age                                                         
U0001  62    301                301          301     301     301   
U0002  26    301                301          301     301     301   
U0003  22    301                301          301     301     301   
U0004  24    301                301          301     301     301   
U0005  34    301                301          301     301     301   

            Most listened musics  Most read books  Most visited websites  \
UserId Age                                                                 
U0001  62                    301              301                    301   
U0002  26                    301              301                    301   
U0003  22                    301              301                    301   
U0004  24                    301              301                    301   
U0005  34                    301              301                    301   

            Most watched movies  Most watched tv programmes  ...  fave8  \
UserId Age                                                   ...          
U0001  62                   301                         301  ...      0   
U0002  26                   301                         301  ...      0   
U0003  22                   301                         301  ...      0   
U0004  24                   301                         301  ...      0   
U0005  34                   301                         301  ...      0   

            fave9  unfave1  unfave2  unfave3  unfave4  unfave5  unfave6  \
UserId Age                                                                
U0001  62       0      301      301      301      301      301        0   
U0002  26       0      301      301      301      301      301        0   
U0003  22       0      301      301      301      301      301        0   
U0004  24       0      301      301      301      301      301        0   
U0005  34       0      301      301      301      301      301        0   

            AdFilePath  Rating  
UserId Age                      
U0001  62          301     301  
U0002  26          301     301  
U0003  22          301     301  
U0004  24          301     301  
U0005  34          301     301  

[5 rows x 31 columns]

In [13]:
df_age.groupby(['Age']).count()

AdId  Countries visited  Fave Sports  Gender  Income  \
Age                                                         
18      7                  7            7       7       7   
19      5                  5            5       5       5   
20      4                  4            4       4       4   
21      6                  6            6       6       6   
22      9                  9            9       9       9   
23      4                  4            4       4       4   
24      6                  6            6       6       6   
25      5                  5            5       5       5   
26      6                  6            6       6       6   
27      3                  3            3       3       3   
28      5                  5            5       5       5   
29      1                  1            1       1       1   
30      5                  5            5       5       5   
31      4                  4            4       4       4   
32      6                  6            6       6       6   
33      5                  5            5       5       5   
34      4                  4            4       4       4   
35      4                  4            4       4       4   
36      2                  2            2       2       2   
37      2                  2            2       2       2   
38      2                  2            2       2       2   
39      2                  2            2       2       2   
40      2                  2            2       2       2   
41      2                  2            2       2       2   
43      1                  1            1       1       1   
46      1                  1            1       1       1   
47      1                  1            1       1       1   
48      1                  1            1       1       1   
49      2                  2            2       2       2   
52      2                  2            2       2       2   
55      2                  2            2       2       2   
56      1                  1            1       1       1   
57      2                  2            2       2       2   
61      2                  2            2       2       2   
62      1                  1            1       1       1   
64      1                  1            1       1       1   
66      1                  1            1       1       1   
68      1                  1            1       1       1   

     Most listened musics  Most read books  Most visited websites  \
Age                                                                 
18                      7                7                      7   
19                      5                5                      5   
20                      4                4                      4   
21                      6                6                      6   
22                      9                9                      9   
23                      4                4                      4   
24                      6                6                      6   
25                      5                5                      5   
26                      6                6                      6   
27                      3                3                      3   
28                      5                5                      5   
29                      1                1                      1   
30                      5                5                      5   
31                      4                4                      4   
32                      6                6                      6   
33                      5                5                      5   
34                      4                4                      4   
35                      4                4                      4   
36                      2                2                      2   
37                      2                2                      2   
38                      2                2    

In [43]:
df_income = df.groupby(['UserId', 'Income']).count()
df_income

AdId  Age  Countries visited  Fave Sports  Gender  \
UserId Income                                                      
U0001  0          0    0                  0            0       0   
       1        301  301                301          301     301   
       2          0    0                  0            0       0   
       3          0    0                  0            0       0   
U0002  0          0    0                  0            0       0   
...             ...  ...                ...          ...     ...   
U0119  3          0    0                  0            0       0   
U0120  0          0    0                  0            0       0   
       1          0    0                  0            0       0   
       2        301  301                301          301     301   
       3          0    0                  0            0       0   

               Most listened musics  Most read books  Most visited websites  \
UserId Income                                                                 
U0001  0                          0                0                      0   
       1                        301              301                    301   
       2                          0                0                      0   
       3                          0                0                      0   
U0002  0                          0                0                      0   
...                             ...              ...                    ...   
U0119  3                          0                0                      0   
U0120  0                          0                0                      0   
       1                          0                0                      0   
       2                        301              301                    301   
       3                          0                0                      0   

               Most watched movies  Most watched tv programmes  ...  fave8  \
UserId Income                                                   ...          
U0001  0                         0                           0  ...      0   
       1                       301                         301  ...      0   
       2                         0                           0  ...      0   
       3                         0                           0  ...      0   
U0002  0                         0                           0  ...      0   
...                            ...                         ...  ...    ...   
U0119  3                         0                           0  ...      0   
U0120  0                         0                           0  ...      0   
       1                         0                           0  ...      0   
       2                       301                         301  ...      0   
       3                         0                           0  ...      0   

               fave9  unfave1  unfave2  unfave3  unfave4  unfave5  unfave6  \
UserId Income                                                                
U0001  0           0        0        0        0        0        0        0   
       1           0      301      301      301      301      301        0   
       2           0        0        0        0        0        0        0   
       3           0        0        0        0        0        0        0   
U0002  0           0        0        0        0        0        0        0   
...              ...      ...      ...      ...      ...      ...      ...   
U0119  3           0        0        0        0        0        0        0   
U0120  0           0        0        0        0        0        0        0   
       1           0        0        0        0        0        0        0   
       2           0      301      301      301      301      301        0   
       3           0        0        0        0        0        0        0   

               AdFilePath  Rating  
UserId Income                      
